x-pack/plugins/lens/public/xy_visualization/axis_settings_popover.tsx

```diff
@@ -236,6 +286,178 @@ export const AxisSettingsPopover: React.FunctionComponent<AxisSettingsPopoverPro
+                    })}
+                    isInvalid={inclusiveZeroError || boundaryError}
+                    helpText={
+                      hasBarOrAreaOnAxis && !inclusiveZeroError
+                        ? i18n.translate('xpack.lens.xyChart.inclusiveZero', {
+                            defaultMessage:
+                              'Bounds have to include include zero for area and bar series',
+                          })
+                        : undefined
+                    }
+                    error={
+                      hasBarOrAreaOnAxis && inclusiveZeroError
+                        ? i18n.translate('xpack.lens.xyChart.inclusiveZero', {
+                            defaultMessage:
+                              'Bounds have to include include zero for area and bar series',
+                          })
+                        : undefined
+                    }
+                  >
+                    <EuiFieldNumber
+                      value={localExtent.lowerBound ?? ''}
+                      isInvalid={inclusiveZeroError || boundaryError}
+                      data-test-subj="lnsXY_axisExtent_lowerBound"
+                      onChange={(e) => {
+                        const val = Number(e.target.value);
+                        if (e.target.value === '' || Number.isNaN(Number(val))) {
+                          setLocalExtent({
+                            ...localExtent,
+                            lowerBound: undefined,
+                          });
+                        } else {
+                          setLocalExtent({
+                            ...localExtent,
+                            lowerBound: val,
+                          });
+                        }
+                      }}
+                      onBlur={() => {
+                        if (localExtent.lowerBound === undefined && dataBounds) {
+                          setLocalExtent({
+                            ...localExtent,
+                            lowerBound: Math.min(0, dataBounds.min),
+                          });
+                        }
+                      }}
+                    />
+                  </EuiFormRow>
+                </EuiFlexItem>
+                <EuiFlexItem>
+                  <EuiFormRow
+                    display="rowCompressed"
+                    fullWidth
+                    label={i18n.translate('xpack.lens.xyChart.upperBoundLabel', {
+                      defaultMessage: 'Upper bound',
+                    })}
+                    isInvalid={inclusiveZeroError || boundaryError}
+                    error={
+                      boundaryError
+                        ? i18n.translate('xpack.lens.xyChart.boundaryError', {
+                            defaultMessage: 'Lower bound has to be larger than upper bound',
+                          })
+                        : undefined
+                    }
+                  >
+                    <EuiFieldNumber
```

Comment:
`EuiFieldNumber` needs a `compressed` prop in order to apply the compressed styles.